### PR TITLE
Chore: Upgrade golangci-lint to v1.51.2

### PR DESCRIPTION
**Pull request by sakjur**
Continuation of #63578 by @papagian.

- We need a later version of golangci-lint because v1.50.1 consumes all memory before crashing on Go 1.20.
- Disabling sqlclosecheck, while the intent of the linter is very good, the linter isn't sufficient to detect the behavior it's supposed to prevent, as discussed in https://github.com/grafana/grafana/pull/63578#discussion_r1114683301 .
- Various changes to accomodate to linters, plus some that were detected by sqlclosecheck, but as per the discussion linked above, I've chosen to avoid defers when we'd want to return the error rather than just log it.

I'm not backporting this despite intending to backport the upgrade of Go and therefore need similar PRs targeting the old releases, simply because I think resolving the merge conflicts in backporting will be more work than opening separate PRs to handle the legacy versions.

### CODE REVIEW

File: pkg/tsdb/cloudmonitoring/utils.go [hunk 1]
```diff
@@ -86,6 +87,11 @@ func doRequestPage(ctx context.Context, logger log.Logger, r *http.Request, dsIn
 		return cloudMonitoringResponse{}, err
 	}
 
+	defer func() {
+		err := res.Body.Close()
+		logger.Warn("failed to close response body", "error", err)
```
Comment (papagian) [inline pkg/tsdb/cloudmonitoring/utils.go:92]: just checking if it's it on purpose `Warn` instead of `Error`

Comment (sakjur) [reply pkg/tsdb/cloudmonitoring/utils.go:92]: Yes, I don't think it's necessarily actionable that a response body wasn't properly closed. If it happens a lot it's a problem, if it happens once or twice, meh.

Comment (papagian) [reply pkg/tsdb/cloudmonitoring/utils.go:92]: 👍 should we change it then for the rest as well?

Comment (sakjur) [reply pkg/tsdb/cloudmonitoring/utils.go:92]: hm, yeah

**Review by papagian — APPROVED** [review]

**Pull request merged**